src/Temporalio/Bridge/src/client.rs

```diff
@@ -0,0 +1,453 @@
+        "RemoveSearchAttributes" => rpc_call!(client, call, remove_search_attributes),
+        rpc => Err(anyhow::anyhow!("Unknown RPC call {}", rpc)),
+    }
+}
+
+async fn call_test_service(client: &CoreClient, call: &RpcCallOptions) -> anyhow::Result<Vec<u8>> {
+    let rpc = call.rpc.to_str();
+    let mut client = client.clone();
+    match rpc {
+        "GetCurrentTime" => rpc_call!(client, call, get_current_time),
+        "LockTimeSkipping" => rpc_call!(client, call, lock_time_skipping),
+        "SleepUntil" => rpc_call!(client, call, sleep_until),
+        "Sleep" => rpc_call!(client, call, sleep),
+        "UnlockTimeSkippingWithSleep" => rpc_call!(client, call, unlock_time_skipping_with_sleep),
+        "UnlockTimeSkipping" => rpc_call!(client, call, unlock_time_skipping),
+        rpc => Err(anyhow::anyhow!("Unknown RPC call {}", rpc)),
+    }
+}
+
+async fn call_health_service(
+    client: &CoreClient,
+    call: &RpcCallOptions,
+) -> anyhow::Result<Vec<u8>> {
+    let rpc = call.rpc.to_str();
+    let mut client = client.clone();
+    match rpc {
+        "Check" => rpc_call!(client, call, check),
+        rpc => Err(anyhow::anyhow!("Unknown RPC call {}", rpc)),
+    }
+}
+
+fn rpc_req<P: prost::Message + Default>(
+    call: &RpcCallOptions,
+) -> anyhow::Result<tonic::Request<P>> {
+    let proto = P::decode(call.req.to_slice())?;
+    let mut req = tonic::Request::new(proto);
+    if call.metadata.size > 0 {
+        for (k, v) in call.metadata.to_str_map_on_newlines() {
+            req.metadata_mut()
+                .insert(MetadataKey::from_str(k)?, v.parse()?);
+        }
+    }
+    if call.timeout_millis > 0 {
+        req.set_timeout(Duration::from_millis(call.timeout_millis.into()));
+    }
+    Ok(req)
+}
+
+fn rpc_resp<P>(res: Result<tonic::Response<P>, tonic::Status>) -> anyhow::Result<Vec<u8>>
+where
+    P: prost::Message,
+    P: Default,
+{
+    Ok(res?.get_ref().encode_to_vec())
+    // TODO(cretz): Better error encoding
+    // match res {
+    //     Ok(resp) => Ok(resp.get_ref().encode_to_vec()),
+    //     Err(err) => {
+    //         Err(Python::with_gil(move |py| {
+    //             // Create tuple of "status", "message", and optional "details"
+    //             let code = err.code() as u32;
+    //             let message = err.message().to_owned();
+    //             let details = err.details().into_py(py);
+    //             RPCError::new_err((code, message, details))
+    //         }))
+    //     }
+    // }
+}
+
+impl TryFrom<&ClientOptions> for CoreClientOptions {
+    type Error = anyhow::Error;
+
+    fn try_from(opts: &ClientOptions) -> anyhow::Result<Self> {
+        let mut opts_builder = ClientOptionsBuilder::default();
+        opts_builder
+            .target_url(Url::parse(opts.target_url.to_str())?)
+            .client_name(opts.client_name.to_str())
+            .client_version(opts.client_version.to_str())
+            .identity(opts.identity.to_string())
```

Comment:
It's a bit odd that two of these are `str` and one is `String`. FYI the first two are just going to get converted into `String`s straight away.

Reply:
`client_name` and `client_version` here take an `Into<String>` and `identity` here takes a `String` only so I figured I'd give each they lowest level they can accept.

Reply:
It's not really about what's "lowest-level", it's about these lines implying that this is going to get borrowed when really it's getting cloned. The builder accepts into for `&'static str`, mainly, for convenience. 

It's not a big deeal, but just letting you know about a best practice.

Reply:
Fixed